### ISSUE TITLE
Where can I find the weight of the trained nnU-Net?

### ISSUE THREAD

**Issue by WanzeeCho** [MIC-DKFZ/nnUNet#2240]
Since I did not receive a sufficient answer to my previous question (#2233), I asked additional questions but did not receive an answer. Therefore, I have created a new issue, so please understand.

Here's what I'm trying to do:

1. Pre-train nnU-Net using T1 images and labels
2. Initialize nnU-Net (Teacher Model) with weights of '1'
3. Train '2' on unlabeled T2 data
4. Make pseudo-label from Teacher Model (To increase the accuracy of the pseudo-label, we want to import weights from the pre-trained model)

In this case, what is the path of weights of '1'?
Or is there another way to pre-train? In order to proceed according to the https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/pretraining_and_finetuning.md document provided, a target dataset is required, but in my case, there is no target dataset.

I have already completed learning with the following commands using the T1 image.

```bash
CUDA_VISIBLE_DEVICES=4 nnUNetv2_train DATASET_ID 3d_fullres 0 --npz
CUDA_VISIBLE_DEVICES=4 nnUNetv2_train DATASET_ID 3d_fullres 1 --npz
CUDA_VISIBLE_DEVICES=5 nnUNetv2_train DATASET_ID 3d_fullres 2 --npz
CUDA_VISIBLE_DEVICES=5 nnUNetv2_train DATASET_ID 3d_fullres 3 --npz
CUDA_VISIBLE_DEVICES=4 nnUNetv2_train DATASET_ID 3d_fullres 4 --npz
```
I found the checkpoint_final.pth file for each fold in ../nnUNetTrainer__nnUNetPlans__3d_fullres/fold_x. Is it possible to ensemble this file to obtain the weight of the pre-trained model? (If possible, please also tell me how)

Additionally, I plan to use this weight in the student model as well.

**Comment by WanzeeCho** [comment]
Through this code, the weight of each fold was made into one weight.


```bash
import torch

checkpoints = []
for fold in range(5):
 checkpoint = torch.load(f'../nnUNetv2/nnUNet_results/Dataset825_ceT1_pretrain/nnUNetTrainer__nnUNetPlans__3d_fullres/fold_{fold}/checkpoint_final.pth')
 checkpoints.append(checkpoint['network_weights'])
 # dict_keys(['network_weights', 'optimizer_state', 'grad_scaler_state', 'logging', '_best_ema', 'current_epoch', 'init_args', 'trainer_name', 'inference_allowed_mirroring_axes'])

avg_state_dict = {}
for key in checkpoints[0].keys():
 avg_state_dict[key] = torch.mean(torch.stack([checkpoints[fold][key].float() for fold in range(5)]), dim=0)

torch.save(avg_state_dict, 'ensemble_model_final_checkpoint.pth')
```

I'm going to use these weights wherever I want.

Hoping to use the weights of the learned nnU-Net, I obtained pseudo-labels by inserting unlabeled T2 data into the learned nnU-Net.

**Comment by WanzeeCho** [comment]
Soved
